sdk/storage/Azure.Storage.Files.DataLake/src/Models/PathAccessControlEntry.cs

```diff
@@ -0,0 +1,207 @@
+        /// Indicates whether this is the default entry for the ACL.
+        /// </summary>
+        public bool DefaultScope { get; set; }
+
+        /// <summary>
+        /// Specifies which role this entry targets.
+        /// </summary>
+        public AccessControlType AccessControlType { get; set; }
+
+        /// <summary>
+        /// Specifies the entity for which this entry applies.
+        /// Must be omitted for types mask or other.  It must also be omitted when the user or group is the owner.
+        /// </summary>
+        public string EntityId { get; set; }
+
+        /// <summary>
+        /// Specifies the permissions granted to this entry.
+        /// </summary>
+        public RolePermissions Permissions { get; set; }
+
+        /// <summary>
+        /// Empty constructor.
+        /// </summary>
+        public PathAccessControlEntry() { }
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="accessControlType">Specifies which role this entry targets.</param>
+        /// <param name="permissions">Specifies the permissions granted to this entry.</param>
+        /// <param name="defaultScope">Indicates whether this is the default entry for the ACL.</param>
+        /// <param name="entityId">Specifies the entity for which this entry applies.</param>
+        public PathAccessControlEntry(
+            AccessControlType accessControlType,
+            RolePermissions permissions,
+            bool defaultScope = false,
+            string entityId = null)
+        {
+            DefaultScope = defaultScope;
+            AccessControlType = accessControlType;
+            EntityId = entityId;
+            Permissions = permissions;
+        }
+
+        /// <summary>
+        /// Override of ToString()
+        /// </summary>
+        /// <returns></returns>
+        public override string ToString()
+        {
+            StringBuilder stringBuilder = new StringBuilder();
+
+            if (DefaultScope)
+            {
+                stringBuilder.Append("default:");
+            }
+            stringBuilder.Append(AccessControlType.ToString().ToLowerInvariant());
+            stringBuilder.Append(":");
+            stringBuilder.Append(EntityId ?? "");
+            stringBuilder.Append(":");
+            stringBuilder.Append(Permissions.ToSymbolicString());
+
+            return stringBuilder.ToString();
+        }
+
+        /// <summary>
+        /// Parses the provided string into a <see cref="PathAccessControlEntry"/>
+        /// </summary>
+        /// <param name="str">The string representation of the ACL.</param>
+        /// <returns>A <see cref="PathAccessControlEntry"/>.</returns>
+        public static PathAccessControlEntry Parse(string str)
+        {
+            PathAccessControlEntry entry = new PathAccessControlEntry();
+            string[] parts = str.Split(':');
+            int indexOffset = 0;
+
+            if (parts.Length == 4)
```

Comment:
add 4 to Constants

Reply:
There's a lot of strings and characters floating around PathAccessControlEntry and RolePermissionsExtensions.  Do you think I should also move them?

Reply:
I think all these values are pretty specific to the format being parsed and not sure they benefit much from being moved into Constants.cs